File: services/Roblox/Roblox.Website/Controllers/v1/AccountSettings.cs
using System.ComponentModel.DataAnnotations;
using Dapper;
using Microsoft.AspNetCore.Mvc;
using Roblox.Website.WebsiteModels;

namespace Roblox.Website.Controllers;

[ApiController]
[Route("/apisite/accountsettings/v1")]
public class AccountSettingsControllerV1 : ControllerBase
{
    [HttpGet("email")]
    public dynamic GetEmail()
    {
        return new
        {
            emailAddress = "info@economysimulator.com",
            verified = true,
        };
    }

    [HttpGet("themes/types")]
    public dynamic GetThemes()
    {
        return new
        {
            data = Enum.GetNames<Models.Users.ThemeTypes>(),
        };
    }

    [HttpGet("themes/user")]
    public async Task<dynamic> GetUserTheme()
    {
        var settings = await services.accountInformation.GetUserTheme(safeUserSession.userId);
        return new
        {
            themeType = settings,
        };
    }

    [HttpPatch("themes/user")]
    public async Task SetUserTheme([Required, FromBody] SetThemeRequest request)
    {
        await services.accountInformation.SetUserTheme(safeUserSession.userId, request.themeType);
    }

    [HttpGet("content-restriction")]
    public dynamic GetContentRestriction()
    {
        return new
        {
            contentRestrictionLevel = "NoRestrictions",
        };
    }

    [HttpGet("inventory-privacy")]
    public async Task<dynamic> GetInventoryPrivacy()
    {
        return await services.accountInformation.GetUserInventoryPrivacy(safeUserSession.userId);
    }

    [HttpPost("inventory-privacy")]
    public async Task SetInventoryPrivacy([Required, FromBody] SetInventoryPrivacyRequest request)
    {
        await services.accountInformation.SetUserInventoryPrivacy(safeUserSession.userId, request.inventoryPrivacy);
    }

    [HttpGet("trade-privacy")]
    public async Task<dynamic> GetTradePrivacy()
    {
        var res = await services.accountInformation.GetUserTradePrivacy(safeUserSession.userId);
        return new
        {
            tradePrivacy = res,
        };
    }
    
    [HttpPost("trade-privacy")]
    public async Task SetTradePrivacy([Required, FromBody] SetTradePrivacyRequest request)
    {
        await services.accountInformation.SetUserTradePrivacy(safeUserSession.userId, request.tradePrivacy);
    }

    [HttpGet("trade-value")]
    public async Task<dynamic> GetTradeValue()
    {
        var res = await services.accountInformation.GetUserTradeValue(safeUserSession.userId);
        return new
        {
            tradeValue = res,
        };
    }

    [HttpPost("trade-value")]
    public async Task SetTradeValue([Required, FromBody] SetTradeValueRequest request)
    {
        await services.accountInformation.SetUserTradeValue(safeUserSession.userId, request.tradeValue);
    }
}

File: services/2016-roblox-main/services/accountSettings.js
import request, { getBaseUrl, getFullUrl } from "../lib/request"

export const getMySettingsJson = () => {
  return request('GET', getBaseUrl() + '/my/settings/json').then(d => d.data);
}

export const getMyEmail = () => {
  return request('GET', getFullUrl('accountsettings', '/v1/email')).then(d => d.data);
}

export const getInventoryPrivacy = () => {
  return request('GET', getFullUrl('accountsettings', '/v1/inventory-privacy')).then(d => d.data);
}

export const setInventoryPrivacy = ({ newPrivacy }) => {
  return request('POST', getFullUrl('accountsettings', '/v1/inventory-privacy'), {
    inventoryPrivacy: newPrivacy,
  });
}

export const getTradePrivacy = () => {
  return request('GET', getFullUrl('accountsettings', '/v1/trade-privacy')).then(d => d.data.tradePrivacy);
}

export const setTradePrivacy = ({ newPrivacy }) => {
  return request('POST', getFullUrl('accountsettings', '/v1/trade-privacy'), {
    tradePrivacy: newPrivacy,
  });
}

export const getTradeValue = () => {
  return request('GET', getFullUrl('accountsettings', '/v1/trade-value')).then(d => d.data.tradeValue);
}

export const setTradeValue = ({ newValue }) => {
  return request('POST', getFullUrl('accountsettings', '/v1/trade-value'), {
    tradeValue: newValue,
  });
}